src/healthData/HealthDataManager.js

```diff
@@ -0,0 +1,180 @@
+            guid = uuid.v4();
+            PreferencesManager.setViewState("GUID", guid);
+        }
+
+        oneTimeHealthData.guid = guid;
+        oneTimeHealthData.snapshotTime = (new Date()).getTime();
+        oneTimeHealthData.os = brackets.platform;
+        oneTimeHealthData.osVersion = HealthDataUtils.getOSVersion();
+        oneTimeHealthData.osLanguage = brackets.app.language;
+        oneTimeHealthData.bracketsLanguage = brackets.getLocale();
+        oneTimeHealthData.bracketsVersion = brackets.metadata.version;
+
+        HealthDataUtils.getInstalledExtensions()
+            .done(function (userInstalledExtensions) {
+                oneTimeHealthData.installedExtensions = userInstalledExtensions;
+            })
+            .always(function () {
+                return result.resolve(oneTimeHealthData);
+            });
+
+        return result.promise();
+    }
+
+    /**
+     * Send data to the server
+     */
+    function sendHealthDataToServer() {
+        var result = new $.Deferred();
+        
+        getHealthData().done(function (jsonData) {
+            var url;
+            if (params.get("testEnvironment")) {
+                url = brackets.config.healthDataTestServerURL;
+            } else {
+                url = brackets.config.healthDataServerURL;
+            }
+
+            $.ajax({
+                url: url,
+                type: "POST",
+                data: jsonData,
+                dataType: "text",
+                contentType: "text/plain"
+            })
+                .done(function () {
+                    result.resolve();
+                })
+                .fail(function (jqXHR, status, errorThrown) {
+                    console.error("Error in sending health data. Response : " + jqXHR.responseText + ". Status : " + status + ". Error : " + errorThrown);
+                    result.reject();
+                })
+                .always(function () {
+                    PreferencesManager.setViewState("lastTimeSendHealthData", (new Date()).getTime());
+                });
+        })
+            .fail(function () {
+                result.reject();
+            });
+
+        return result.promise();
+    }
+
+    /*
+     * Check if health data is to be send to the server. If user has enable the tracking, health data will be send for every 24 hours.
+     * If 24 hours or more than that has been passed, then send health data to the server
+     */
+    function _checkHealthDataExport() {
+        var result = new $.Deferred();
+        var isHDTracking = prefs.get("healthDataTracking");
+        window.clearTimeout(timeoutVar);
+
+        if (isHDTracking) {
+
+            var lastTimeSend = PreferencesManager.getViewState("lastTimeSendHealthData"),
+                currentTime = (new Date()).getTime();
+
+            if (!lastTimeSend) {
+                var randomTime = Math.floor(Math.random() * ONE_DAY);
+                lastTimeSend = currentTime + randomTime;
```

Comment:
@peterflynn Initially we decided to have random time to distribute load on server. But @abose @prksingh have tested the load on server and it can easily handle requests. So no randomization needed now.